File: leetcode/go/118.pascals-triangle.go
func generate(numRows int) [][]int {
    pt := make([][]int, numRows)
    row := []int{1}
    newRow := []int{}
    pt[0] = row
    for i:=1; i<numRows; i++ {
        newRow = make([]int, i+1)
        newRow[0] = 1
        j := 1
        for ; j < i; j++ {
            newRow[j] = row[j] + row[j-1]
        }
        newRow[j] = 1
        pt[i] = newRow
        row = newRow
    }
    return pt
}

File: leetcode/rust-leetcode/src/118.pascals-triangle.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut pt = Vec::with_capacity(num_rows as usize);
        let mut row = vec![1];
        pt.push(row.to_vec());
        let mut new_row = vec![];
        for _ in 1..num_rows {
            new_row.push(1);
            for j in 1..row.len() {
                new_row.push(row[j] + row[j-1]);
            }
            new_row.push(1);
            pt.push(new_row.to_vec());
            row = new_row.to_vec();
            new_row.clear();
        }
        pt
    }
}

